inbox/models/message.py

```diff
@@ -255,7 +259,11 @@ def create_from_synced(cls, account, mid, folder_name, received_date,
         msg.data_sha256 = sha256(body_string).hexdigest()
 
         # Persist the raw MIME message to disk/ S3
-        save_to_blockstore(msg.data_sha256, body_string)
+        encrypted_body_string = vault_encrypt(
+            body_string,
+            account.namespace.public_id
+        )
+        save_to_blockstore(msg.data_sha256, encrypted_body_string)
```

Comment:
Was it already encrypted by nylas? or what msg.data_sha256 means here?

Reply:
It's not encrypted by Nylas. It's encrypted by us using Vault. `save_to_blockstore` just saves message content to local file or S3 (depends on configuration).
`data_sha256` - is just a checksum. It's used for skipping the same messages from saving.